galileo/examples/highlight_features.rs

```diff
@@ -0,0 +1,181 @@
+    ) -> bool {
+        self.point.is_point_inside(point, tolerance)
+    }
+
+    fn bounding_rectangle(
+        &self,
+    ) -> Option<
+        galileo_types::cartesian::Rect<
+            <Point2d as galileo_types::cartesian::CartesianPoint2d>::Num,
+        >,
+    > {
+        None
+    }
+}
+
+struct ColoredPointSymbol {}
+
+pub async fn run(builder: MapBuilder) {
+    #[cfg(not(target_arch = "wasm32"))]
+    let builder = builder.with_raster_tiles(
+        |index| {
+            format!(
+                "https://tile.openstreetmap.org/{}/{}/{}.png",
+                index.z, index.x, index.y
+            )
+        },
+        TileSchema::web(18),
+    );
+
+    let projection = Crs::EPSG3857
+        .get_projection()
+        .expect("must find projection");
+
+    let feature_layer = FeatureLayer::new(
+        [
+            latlon!(53.732562, -1.863383),
+            latlon!(53.728265, -1.839966),
+            latlon!(53.704014, -1.786128),
+        ]
+        .iter()
+        .map(|point| PointMarker {
+            point: projection.project(point).unwrap(),
+            ..Default::default()
+        })
+        .collect(),
+        ColoredPointSymbol {},
+        Crs::EPSG3857,
+    );
+
+    let feature_layer = Arc::new(RwLock::new(feature_layer));
+
+    builder
+        .center(latlon!(53.732562, -1.863383))
+        .resolution(30.0)
+        .with_layer(feature_layer.clone())
+        .with_event_handler(move |ev, map| {
+            if let UserEvent::PointerMoved(event) = ev {
+                let mut layer = feature_layer.write().unwrap();
+
+                let Some(position) = map.view().screen_to_map(event.screen_pointer_position) else {
+                    return EventPropagation::Stop;
+                };
+
+                for mut feature_container in layer.features_mut().iter_mut() {
+                    feature_container.as_mut().highlighted = false;
+                }
+
+                for mut feature_container in
+                    layer.get_features_at_mut(&position, map.view().resolution() * 20.0)
+                {
+                    let state = feature_container.as_ref().highlighted;
+                    feature_container.as_mut().highlighted = !state;
+                }
+
+                map.redraw();
+            }
+            galileo::control::EventPropagation::Propagate
+        })
+        .build()
+        .await
+        .run();
+}
+
+impl Symbol<PointMarker> for ColoredPointSymbol {
+    fn render<'a, N, P>(
+        &self,
+        feature: &PointMarker,
+        geometry: &'a Geom<P>,
+        _min_resolution: f64,
+    ) -> Vec<RenderPrimitive<'a, N, P, Contour<P>, Polygon<P>>>
+    where
+        N: AsPrimitive<f32>,
+        P: CartesianPoint3d<Num = N> + Clone,
+    {
+        if let Geom::Point(point) = geometry {
+            vec![RenderPrimitive::new_point_ref(
+                point,
+                PointPaint::image(
```

Comment:
`render` function is called for each feature multiple times during lifetime of a layer. So instead of cloning the image many times, the preferred approach should be to create all the `PointPaint` instances when the symbol is created, and then reuse them for all features.

Reply:
Since the images are already within an `Arc`, doesn't `.clone()` on that only increase the reference count as opposed to cloning the actual image?

I tried what you suggested but ended up having to clone the point paint instances as well...maybe I am misunderstanding something here.

Reply:
Yeah, I missed the part where the images are decoded and stored in the static. Sorry for that. It's fine as is then.